### DISCUSSION TITLE
Embed Tweet in next13 appDir

### DISCUSSION THREAD

**Discussion by YoannBuzenet** [vercel/next.js#46774] [Help] [answered]
### Summary

Hello,

There are [answers](https://github.com/vercel/next.js/discussions/15262) about how to implement embedded tweets in nextjs "classic" but not with the `appDir`.

As a reminder, Twitter gives a `html/js` snippet that loads an iframe that displays the tweet.

- Twitter Iframe are loaded correctly when the page is displaying with SSR
- When using the next router with a `Link`, it doesn't load. It remains the snippet given by twitter, even if I preloaded the twitter js widget before. The iframe loading is not triggered.

Would you have an idea on how to do it ? 

Thanks

**Comment by YoannBuzenet** [comment, marked as answer]
I found what was triggering by checking the twitter script `widget.js`.

Preload the script : 

```
React.useEffect(() => {
    const s = document.createElement("script");
    s.setAttribute("src", "https://platform.twitter.com/widgets.js");
    s.setAttribute("async", "true");
    document.head.appendChild(s);
  }, []);
```

Then add this and it will work :

`useEffect(() => {
    window.twttr.widgets.load();
  }, []);`

**Reply by guitheengineer** [reply]
For the first part of your code, you could just have used a script tag.
